consensus/istanbul/core/core.go

```diff
@@ -310,7 +310,13 @@ func (c *core) commit() error {
 			c.waitForDesiredRound(nextRound)
 			return nil
 		}
-		if err := c.backend.Commit(proposal, aggregatedSeal, aggregatedEpochValidatorSetSeal); err != nil {
+
+		result, isCached := c.current.GetBlockConsensusAndProcessResult(proposal.Hash())
+		if !isCached {
+			log.Error("BlockConsensusAndProcessResult not found in cache", "number", proposal.Number(), "hash", proposal.Hash())
+			return nil
```

Comment:
This should return a non nil error.  If the cached entry is not there, then there is a bug in the code.   Please create a new error in `core/errors.go` for that.